### ISSUE TITLE
Uncaught TypeError: Cannot read property 'row' of undefined

### ISSUE THREAD

**Issue by KART35** [atom-community/linter#771]
[Enter steps to reproduce below:]

This is triggered pretty much any time the linter comes back with an error. Happens with both linter-clang and linter-javac.

**Atom Version**: 1.0.2
**System**: Ubuntu 15.04
**Thrown From**: [linter](https://github.com/atom-community/linter) package, v1.2.4

### Stack Trace

Uncaught TypeError: Cannot read property 'row' of undefined

```
At /home/ian/.atom/packages/linter/lib/views/message.coffee:18

TypeError: Cannot read property 'row' of undefined
  at Function.Message.renderLink (/home/ian/.atom/packages/linter/lib/views/message.coffee:18:54)
  at linter-message.Message.attachedCallback (/home/ian/.atom/packages/linter/lib/views/message.coffee:7:26)
  at linter-panel.<anonymous> (/home/ian/.atom/packages/linter/lib/views/bottom-panel.js:41:12)
  at Array.forEach (native)
  at linter-panel.updateMessages (/home/ian/.atom/packages/linter/lib/views/bottom-panel.js:40:14)
  at LinterViews.renderPanelMessages (/home/ian/.atom/packages/linter/lib/linter-views.coffee:93:12)
  at LinterViews.render (/home/ian/.atom/packages/linter/lib/linter-views.coffee:46:6)
  at Commands.lint (/home/ian/.atom/packages/linter/lib/commands.coffee:53:21)
  at atom-workspace.Commands.subscriptions.add.atom.commands.add.linter:lint (/home/ian/.atom/packages/linter/lib/commands.coffee:13:26)
  at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/usr/share/atom/resources/app.asar/src/command-registry.js:241:29)
  at CommandRegistry.handleCommandEvent (/usr/share/atom/resources/app.asar/src/command-registry.js:3:61)
  at CommandRegistry.module.exports.CommandRegistry.dispatch (/usr/share/atom/resources/app.asar/src/command-registry.js:156:19)
  at Object.module.exports.lint (/home/ian/.atom/packages/build-tools/lib/main.coffee:109:19)
  at process.BufferedProcess.exit (/home/ian/.atom/packages/build-tools/lib/main.coffee:143:10)
  at triggerExitCallback (/usr/share/atom/resources/app.asar/src/buffered-process.js:213:47)
  at /usr/share/atom/resources/app.asar/src/buffered-process.js:227:18
  at Socket.<anonymous> (/usr/share/atom/resources/app.asar/src/buffered-process.js:98:18)
  at emitOne (events.js:82:20)
  at Socket.emit (events.js:166:7)
  at Pipe.close (net.js:464:12)

```

### Commands

```
  5x -1:45.1.0 core:backspace (atom-text-editor.editor.is-focused.autocomplete-active)
     -1:33.1.0 docblockr:parse-enter (atom-text-editor.editor.is-focused)
     -1:33.1.0 editor:newline (atom-text-editor.editor.is-focused)
 13x -1:25.3.0 core:backspace (atom-text-editor.editor.is-focused.autocomplete-active)
     -1:12 core:save (atom-text-editor.editor.is-focused)
     -1:11.4.0 build-tools:first-command (atom-text-editor.editor.is-focused)
     -1:10.5.0 linter:lint (atom-workspace.workspace.scrollbars-visible-always.theme-one-dark-syntax.theme-one-dark-ui)
     -0:41.3.0 core:copy (atom-text-editor.editor.is-focused)
  2x -0:40 core:paste (atom-text-editor.editor.is-focused)
     -0:36.1.0 build-tools:first-command (atom-text-editor.editor.is-focused)
     -0:35 linter:lint (atom-workspace.workspace.scrollbars-visible-always.theme-one-dark-syntax.theme-one-dark-ui)
     -0:05.3.0 core:backspace (atom-text-editor.editor.is-focused)
     -0:04.2.0 editor:move-to-end-of-screen-line (atom-text-editor.editor.is-focused)
  2x -0:03.6.0 core:move-left (atom-text-editor.editor.is-focused)
     -0:01.8.0 build-tools:first-command (atom-text-editor.editor.is-focused.autocomplete-active)
     -0:00.6.0 linter:lint (atom-workspace.workspace.scrollbars-visible-always.theme-one-dark-syntax.theme-one-dark-ui)
```

### Config

```json
{
  "core": {},
  "linter": {}
}
```

### Installed Packages

```coffee
# User
atom-beautify, v0.28.8
autocomplete-clang, v0.8.7
autocomplete-paths, v1.0.2
build-tools, v3.0.2
clang-format, v1.14.0
docblockr, v0.7.3
formatter, v2.12.3
git-log, v0.4.1
git-plus, v5.2.2
linter, v1.2.4
linter-clang, v2.26.2
linter-javac, v0.1.4
minimap, v4.12.0

# Dev
No dev packages
```

**Comment by steelbrain** [comment]
Just another issue from a linter using the legacy API. We're gonna remove it anyway, not sure if this bug is valid.
I would advise you open up a bug or do `+1` if you find existing one in the `javac` repo asking the maintainer to upgrade the package to the latest version.
